src/shared/components/tc-communities/JoinCommunity/index.jsx

```diff
@@ -50,23 +51,24 @@ export default function JoinCommunity({
           }
           showJoinConfirmModal();
         }}
-        className={`${theme.link} ${state === STATE.JOINING ? style.joining : ''}`}
+        className={state === STATE.JOINING ? style.joining : ''}
+        theme={{ button: theme.link, disabled: style.disabled }}
```

Comment:
I think, for the Join button itself, it will be better to pass entire theme as a dedication optional prop, say `joinButtonTheme`, and then here we pass into the button `theme={joinButtonTheme}`. It will be always used for ad-hoc correction of Join button style, so it is fine.